packages/ui/src/app/providers/backend/context.ts

```diff
@@ -1,4 +1,17 @@
 import { ApolloClient } from '@apollo/client'
 import { createContext } from 'react'
 
-export const BackendContext = createContext<ApolloClient<any> | undefined>(undefined)
+export type MemberNotificationSettingsData = {
+  hasBeenAskedForEmail?: boolean
+  accessToken?: string
+}
+
+export type MemberNotificationsRecord = Record<string, MemberNotificationSettingsData>
+
+export type BackendContextValue = {
+  backendClient?: ApolloClient<any>
+  notificationsSettingsMap?: MemberNotificationsRecord
+  setMemberSettings: (memberId: string, settings: Partial<MemberNotificationSettingsData>) => void
+}
+
+export const BackendContext = createContext<BackendContextValue | undefined>(undefined)
```

Comment:
Can't it be initialized with:
```suggestion
export const BackendContext = createContext<BackendContextValue>({ setMemberSettings: () => undefined })
```
instead ?